reservation/reservation.go

```diff
@@ -106,6 +106,16 @@ func (manager *Manager) Lock(resource string) (*Reservation, error) {
 	return res, nil
 }
 
+// WaitUntilLock creates a Reservation for `resource`, or waits until it can do so.
+func (manager *Manager) WaitUntilLock(resource string) (*Reservation, error) {
+	res, err := manager.Lock(resource)
+	for err != nil && err.Error() == fmt.Sprintf("Reservation already exists for resource %s", resource) {
```

Comment:
Also, instead of doing:

```go
err != nil && err.Error() == fmt.Sprintf("Reservation already exists for resource %s", resource)
```

You can do:

```go
fmt.Sprintf("%s", err) == fmt.Sprintf("Reservation already exists for resource %s", resource)
```